CONTRIBUTING.md

```diff
@@ -1,47 +1,160 @@
+To contribute code or documentation to this project, follow the [fork and pull request](https://docs.github.com/en/get-started/quickstart/contributing-to-projects) workflow.
 
-## Reporting bugs or suggesting improvements
+## Install Langflow from source
 
-Our [GitHub issues](https://github.com/langflow-ai/langflow/issues) page is kept up to date
-with bugs, improvements, and feature requests. There is a taxonomy of labels to help
-with sorting and discovery of issues of interest. [See this page](https://github.com/langflow-ai/langflow/labels) for an overview of
-the system we use to tag our issues and pull requests.
+Install Langflow from source by forking the repository, and then set up your development environment using Make.
 
-If you're looking for help with your code, consider posting a question on the
-[GitHub Discussions board](https://github.com/langflow-ai/langflow/discussions). Please
-understand that we won't be able to provide individual support via email. We
-also believe that help is much more valuable if it's **shared publicly**,
-so that more people can benefit from it.
+### Prerequisites
 
-- **Describing your issue:** Try to provide as many details as possible. What
-  exactly goes wrong? _How_ is it failing? Is there an error?
-  "XY doesn't work" usually isn't that helpful for tracking down problems. Always
-  remember to include the code you ran and if possible, extract only the relevant
-  parts and don't just dump your entire script. This will make it easier for us to
-  reproduce the error.
+* [uv(>=0.4)](https://docs.astral.sh/uv/getting-started/installation/)
+* [Node.js](https://nodejs.org/en/download/package-manager)
+* [Make](https://www.gnu.org/software/make/#documentation)
 
-- **Sharing long blocks of code or logs:** If you need to include long code,
-  logs or tracebacks, you can wrap them in `<details>` and `</details>`. This
-  [collapses the content](https://developer.mozilla.org/en/docs/Web/HTML/Element/details)
-  so it only becomes visible on click, making the issue easier to read and follow.
+### Clone the Langflow repository
 
-## Contributing code and documentation
+1. Navigate to the [Langflow GitHub repository](https://github.com/langflow-ai/langflow), and then click **Fork**.
 
-You can develop Langflow locally and contribute to the Project!
+2. Add the new remote to your local repository on your local machine:
+Replace `YOUR_FORK_NAME` with a name for your fork.
+Replace `YOUR_GIT_USERNAME` with your Git username.
+```bash
+git remote add YOUR_FORK_NAME https://github.com/YOUR_GIT_USERNAME/langflow.git
+```
 
-See [DEVELOPMENT.md](DEVELOPMENT.md) for instructions on setting up and using a development environment.
+### Set up your Langflow development environment
 
-## Opening a pull request
+1. Change your directory to the root of the local Langflow repository:
+```bash
+cd langflow
+```
```

Comment:
I'd kinda hope this step isn't needed but i suppose it makes it more dummy proof